TestCombine.playground/Pages/CreatePublisher.xcplaygroundpage/Contents.swift

```diff
@@ -118,3 +118,105 @@ c1?.cancel()
 //}
 //
 //print(c)
+
+public class Observer<Input, Failure: Error> {
+
+    @Synchronized private var isCancelled: Bool = false
+    private let receiveClosure: (Input) -> Subscribers.Demand
+    private let receiveCompletionClosure: (Subscribers.Completion<Failure>) -> Void
+
+    init(receive: @escaping (Input) -> Subscribers.Demand, receiveCompletion: @escaping (Subscribers.Completion<Failure>) -> Void) {
+        receiveClosure = receive
+        receiveCompletionClosure = receiveCompletion
+    }
+
+    public func receive(_ input: Input) -> Subscribers.Demand {
+        guard isCancelled == false else { return .none }
+        return receiveClosure(input)
+    }
+
+    public func receive(completion: Subscribers.Completion<Failure>) -> Void {
+        guard isCancelled == false else { return }
+        return receiveCompletionClosure(completion)
+    }
+
+    fileprivate func cancel() {
+        isCancelled = true
+    }
+}
+
+public struct Orchestrator {
+    public let stop: () -> Void
+    public let start: (Subscribers.Demand) -> Void
+}
+
+public typealias CreationTask<Input, Failure: Error> = (Observer<Input, Failure>, Subscribers.Demand) -> Orchestrator
+
+public struct Create<Output, Failure: Error>: Publisher {
+
+    private let task: CreationTask<Output, Failure>
+    private let demand: Subscribers.Demand
+
+    public init(maxValues: Subscribers.Demand = .unlimited, task: @escaping CreationTask<Output, Failure>) {
```

Comment:
The subscribing subscriber should choose the demand.
Here is a parameter of the publisher

Reply:
Yeah I agree that this is overkill and it's then received alongside with the observer in the closure. So not really useful since the "creation" block can choice the max values by it's own.
Then the request demand is passed within the other block ... and it's really bad looking
👍

Anyway the goal for me was to followup the workflow